__tests__/app.test.js

```diff
@@ -104,6 +104,55 @@ describe("/api/articles/:article_id/comments", () => {
         expect(body.comments).toEqual([]);
       });
   });
+
+  it("POST:201 responds with the correct comment object", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send({
+        username: "lurker",
+        body: "This is the greatest comment.  EVER.",
+      })
+      .set("Accept", "application/json")
+      .expect("Content-Type", /json/)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toMatchObject({
+          author: "lurker",
+          body: "This is the greatest comment.  EVER.",
+          votes: 0,
+        });
+      });
+  });
+
+  it("POST:400 responds with 'bad request' if provided with invalid article_id", () => {
+    return request(app)
+      .post("/api/articles/chickens/comments")
+      .send({
+        username: "lurker",
+        body: "This is the greatest comment.  EVER.",
+      })
+      .set("Accept", "application/json")
+      .expect("Content-Type", /json/)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+
+  it("POST:404 responds with 'not found' if provided with a non-existent article_id", () => {
+    return request(app)
+      .post("/api/articles/27/comments")
+      .send({
+        username: "lurker",
+        body: "This is the greatest comment.  EVER.",
+      })
+      .set("Accept", "application/json")
+      .expect("Content-Type", /json/)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
```

Comment:
We should also test for the following:

- what happens if there is a missing field in the request body
- what happens if the user does not exists

You may also consider what happens if the request body contains additional fields that are not required. You may consider that the best approach is to ignore these and create the comment anyway.

Reply:
Cool, I've added extra tests to hopefully cover this.  

Regarding the last point on extra fields potentially passed in - as I'm extracting specific data from keys within the insertCommentByArticleId model, any additional body fields will be ignored anyway.